data/SugarBean.php

```diff
@@ -3863,7 +3863,7 @@ public function retrieve($id = -1, $encode = true, $deleted = true)
 
         global $module, $action;
         //Just to get optimistic locking working for this release
-        if ($this->optimistic_lock && $module == $this->module_dir && $action == 'EditView') {
+        if ($this->optimistic_lock && $module == $this->module_dir && $action == 'EditView' && $id == $_REQUEST['record']) {
```

Comment:
It's ok if request has a 'record' but can you please also check for the requested key exists? (for e.g isset($_REQUEST['record']) && $id && ....)